internal/kafka/internal/services/quota.go

```diff
@@ -11,7 +12,7 @@ type QuotaService interface {
 	// CheckIfQuotaIsDefinedForInstanceType checks if quota is defined for the given instance type
 	CheckIfQuotaIsDefinedForInstanceType(kafka *dbapi.KafkaRequest, instanceType types.KafkaInstanceType) (bool, *errors.ServiceError)
 	// ReserveQuota reserves a quota for a user and return the reservation id or an error in case of failure
-	ReserveQuota(kafka *dbapi.KafkaRequest, instanceType types.KafkaInstanceType) (string, *errors.ServiceError)
+	ReserveQuota(kafka *dbapi.KafkaRequest, instanceType types.KafkaInstanceType, sizeRequired int, kafkaConfig *config.KafkaConfig) (string, *errors.ServiceError)
```

Comment:
IIRC, the `kafka *dbapi.KafkaRequest` has the `SizeId`, isn't that enough to infer the `sizeRequired int` info which are passing on as a parameter here?

Reply:
the size has to be processed to get the int value of it

Reply:
Can't you retrieve it from the kafkaConfig?

```go
kafkaConfig.GetKafkaInstanceSize(kafkaRequest.InstanceType, kafkaRequest.SizeId)
```

Reply:
see this comment: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/861#discussion_r827778661

Reply:
removed the params and used kafkaConfig instead